### PR TITLE
Improvement in adding NPL annotation for services

### PR DESCRIPTION
**Pull request by monotosh-avi**
- Bug fix: Skip adding npl annotation for service type NodePort
- UT to verify this fix
- Handling annotation update through status queue

### CODE REVIEW

File: internal/status/status_toiler.go [hunk 1]
```diff
@@ -68,6 +71,12 @@ func DequeueStatus(objIntf interface{}) error {
 		} else if obj.Op == lib.DeleteStatus {
 			DeleteSvcApiGatewayStatusAddress(obj.Options.Key, obj.Options.ServiceMetadata)
 		}
+	} else if obj.ObjType == lib.NPLService {
```
Comment (parimanur) [inline internal/status/status_toiler.go:74]: Nit: Candidate for using switch.

**Review by parimanur — DISMISSED** [review]

File: internal/status/status_toiler.go (hunk 1 above)
Comment (monotosh-avi) [reply internal/status/status_toiler.go]: Added

File: internal/k8s/controller.go [hunk 2]
```diff
@@ -603,7 +603,7 @@ func (c *AviController) SetupEventHandlers(k8sinfo K8sinformers) {
 			}
 			oldobj := old.(*corev1.Service)
 			svc := cur.(*corev1.Service)
-			if oldobj.ResourceVersion != svc.ResourceVersion || reflect.DeepEqual(svc.Annotations, oldobj.Annotations) {
+			if oldobj.ResourceVersion != svc.ResourceVersion || !reflect.DeepEqual(svc.Annotations, oldobj.Annotations) {
```
Comment (sudswasavi) [inline internal/k8s/controller.go:606]: wow - this seems like a big one :)

File: internal/status/svc_annotation.go [hunk 3]
```diff
@@ -38,6 +37,22 @@ func CheckUpdateSvcAnnotation(key, namespace, name string) bool {
...
 	}
+	return false
+}
+
+// UpdateSvcAnnotation updates a Service with NPL annotation, if not already annotated.
+// If the annotation is already pressent return true
+func UpdateNPLAnnotation(key, namespace, name string) {
+	service, err := utils.GetInformers().ServiceInformer.Lister().Services(namespace).Get(name)
+	if err != nil || service.Spec.Type == corev1.ServiceTypeNodePort {
+		return
```
Comment (sudswasavi) [inline internal/status/svc_annotation.go:48]: Can we log here?

**Review by sudswasavi — COMMENTED** [review]
Looks good overall, just a log would complete the PR. Thanks. @monotosh-avi

File: internal/status/svc_annotation.go (hunk 3 above)
Comment (monotosh-avi) [reply internal/status/svc_annotation.go:49]: Added

**Comment by sudswasavi** [comment]
build ako

**Review by sudswasavi — APPROVED** [review]

**Review by sudswasavi — APPROVED** [review]

**Review by hemantavi — APPROVED** [review]

**Pull request merged**